### PR TITLE
composyx: add v1.0.1

### PR DESCRIPTION
**Pull request by fpruvost**
Maphyspp becomes [Composyx](https://gitlab.inria.fr/composyx/composyx).
Move the package.

### CODE REVIEW

File: var/spack/repos/builtin/packages/composyx/package.py [hunk 1]
```diff
@@ -7,16 +7,16 @@
...
 
-    homepage = "https://gitlab.inria.fr/solverstack/maphys/maphyspp"
-    git = "https://gitlab.inria.fr/solverstack/maphys/maphyspp.git"
-    url = "https://gitlab.inria.fr/api/v4/projects/6194/packages/generic/source/v1.1.9/maphyspp-1.1.9.tar.gz"
+    homepage = "https://gitlab.inria.fr/composyx/composyx"
+    git = "https://gitlab.inria.fr/composyx/composyx.git"
+    url = "https://gitlab.inria.fr/api/v4/projects/52455/packages/generic/source/v1.0.1/composyx-1.0.1.tar.gz"
     maintainers("fpruvost")
 
     version("master", branch="master", submodules=True)
```
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/composyx/package.py]: Looks like the "only" branch at https://gitlab.inria.fr/composyx/composyx at this point for the new repo is `main`.

**Review by tldahlgren — CHANGES_REQUESTED** [review]
Confirmed the new homepage and version sha256.  

Looks like the `master` branch has been "replaced" by `main`.

File: var/spack/repos/builtin/packages/composyx/package.py (hunk 1 above)
Comment (fpruvost) [reply var/spack/repos/builtin/packages/composyx/package.py]: Fixed, thanks !

**Review by tldahlgren — APPROVED** [review]
Thanks.

**Pull request merged**